src/common/mfu_daos.c

```diff
@@ -3479,8 +3584,8 @@ static int cont_serialize_props(struct hdf5_args *hdf5,
     }
 
     /* serialize ACL */
-    if (prop_acl != NULL) {
-        entry = &prop_acl->dpp_entries[0];
+    if (prop_query->dpp_nr > 16) {
+        entry = &prop_query->dpp_entries[16];
```

Comment:
@daltonbohning this doesn't totally make sense to me yet. I saw you merged the props and the prop_acl when the properties were retrieved, but it is hard for me to tell if the acl is put into dpp_entries[16].

Reply:
@dsikich I'll leave a comment explaining this